CMake/External_Darknet.cmake

```diff
@@ -0,0 +1,123 @@
+set(allOk True)
+set(errorMessage)
+
+option(AUTO_ENABLE_DARKNET_DEPENDENCY "Automatically turn on all darknet dependencies if darknet is enabled" OFF)
+if(fletch_ENABLE_Darknet AND AUTO_ENABLE_DARKNET_DEPENDENCY)
+  set(dependency OpenCV)
+  set(OneWasOff FALSE)
+
+  foreach (_var IN LISTS dependency)
+    get_property(currentHelpString CACHE "fletch_ENABLE_${_var}" PROPERTY HELPSTRING)
+    set(fletch_ENABLE_${_var} ON CACHE BOOL ${currentHelpString} FORCE)
+    if(NOT TARGET ${_var})
+      include(External_${_var})
+    endif()
+  endforeach()
+endif()
+
+function(addDarknetDendency depend version)
```

Comment:
I did use the External_Caffe as my starting point

All darknet dependencies (OpenCV, CUDA, CUDNN) are optional.

I can streamline this like OpenCV does